### PR TITLE
topdown: In builtin jwt_decode_verify, assert that exp and nbf are number

### PR DESCRIPTION
**Pull request by charlieflowers**
topdown: In builtin jwt_decode_verify, assert that exp and nbf are number if present

Spec allows exp and nbf to be missing, but if they are present, they must be of type "number." Before this change, JWTs that violated this would result in OPA panic. Added assertion on expected type. Added test case for each.

Signed-off-by: Charlie Flowers <cflowers@gmail.com>

### CODE REVIEW

File: topdown/tokens.go
Comment (charlieflowers) [inline topdown/tokens.go:1067]: NOTE: This issue fixes #5165, but I forgot to end the commit message with that!

Let's talk to determine the least painful way to get that resolved depending on whether anyone has pulled my commit.

Comment (anderseknert) [reply topdown/tokens.go:1067]: Making updates in your own fork is OK — you can just amend the commit and push -f 🙂

File: test/cases/testdata/jwtdecodeverify/test-jwtdecodeverify-invalid-nbf-type.yaml
Comment (anderseknert) [inline test/cases/testdata/jwtdecodeverify/test-jwtdecodeverify-invalid-nbf-type.yaml:7]: Maybe have one of the tests take an invalid value other than null, like a string or an object.

**Review by anderseknert — COMMENTED** [review]
Nice! Just update your commit message as you suggested, and maybe consider my suggestion for the test.. but looks good, thanks!

File: topdown/tokens.go
Comment (charlieflowers) [reply topdown/tokens.go:1067]: OK, I'll do this when squashing after PR is approved.

File: test/cases/testdata/jwtdecodeverify/test-jwtdecodeverify-invalid-nbf-type.yaml
Comment (charlieflowers) [reply test/cases/testdata/jwtdecodeverify/test-jwtdecodeverify-invalid-nbf-type.yaml]: Good idea. I changed nbf to pass a string.

File: topdown/tokens.go
Comment (anderseknert) [reply topdown/tokens.go:1067]: LGTM. You can squash right away as that will discard the approval anyway.

**Comment by charlieflowers** [comment]
> 

I'll squash it into one commit and end the message with "Fixes <issue #>". Want me to do that now or wait till you approve the PR?

**Comment by anderseknert** [comment]
@charlieflowers you can go ahead and squash

**Review by anderseknert — APPROVED** [review]

**Pull request merged**
